src/backend/asset-history/index.ts

```diff
@@ -198,5 +199,15 @@ export function createAssetHistorySS({
         }
     }
 
-    messaging.subscribe(TOPIC, WaitLoop);
+    messaging.subscribe(TOPIC, (err, data) => {
+        if (err) {
+            logger.publishLog(LogLevels.ERROR, 'Subscribe failed for: ', SERVICE_INSTANCE_ID, ': ', data);
+            resp.error(data);
+        }
+        if (!ClearBlade.isEdge()) {
```

Comment:
Could we add this check to the initial subscribe? subscribing inside a subscribe callback seems weird to me. Something like - 

```javascript
const theTopic = !ClearBlade.isEdge() ? `${TOPIC}/_platform` : TOPIC;
messaging.subscribe(theTopic (err, data) => {
```

Reply:
We need to be able to listen on both topics because we could have devices publishing directly to the platform (nimbelink) and devices (via edge) publishing to the relay topic:

_monitor/_asset/${ASSETID}/location AND
_monitor/_asset/${ASSETID}/location/_platform

I researched and there is no way to specify a wildcard to handle both topics.

Reply:
Got it. Do you mind factoring this logic out into a util function that can be used by all these files? A comment on the util function would be great as well

Reply:
The logic has been refactored to use promises in the same way RulesEngine uses them.